### ISSUE TITLE
PMC Tools: Logging information on the startup constructor throws

### ISSUE THREAD

**Issue by mdarefull** [aspnet/EntityFramework#8163]
I'm working on an ASP NET MVC Core project (VS 2017, .NET Core 1.1).
I had the need of logging to the console the running environment of the application at startup and hence I configured the loggerFactory on the Startup's constructor and, immediately after, I log the required information:

```c#
public Startup(IHostingEnvironment env, ILoggerFactory loggerFactory)
 {
    //...
    loggerFactory.AddConsole(Configuration.GetSection("Logging"));
    loggerFactory.AddDebug();

    var logger = loggerFactory.CreateLogger<Startup>();
    logger.LogInformation($"Runing in environment: {env.EnvironmentName}");
}
```

Then, if I tried to use a command related to migrations (Add-Migration or Update-Database) I receive the following error:
![image](https://cloud.githubusercontent.com/assets/22510674/25025030/a3f30956-206e-11e7-94f9-7042b24d6206.png)

Furthermore, I noticed that if I don't include the ':' on the message, the command will run successfully:
![image](https://cloud.githubusercontent.com/assets/22510674/25025196/452757dc-206f-11e7-950c-a6440171f2e2.png)

Anyway, the output message is not the expected.

PS: I've superfitially checked by writing directly to the console (Console.WriteLine(...)) and it worked.

### Steps to reproduce
This can be reproduced with the default project templates that come with VS 2017.

1. Create a new solution/project by selecting the template Visual C#/Web/ASP.NET Core Web Application (.NET Core) and choosing the Individual User Accounts authentication.
2. Modify the Startup constructor signature to be:
```c#
public Startup(IHostingEnvironment env, ILoggerFactory loggerFactory)
```
3. At the end of the Startup constructor add the following code:
```c#
loggerFactory.AddConsole(Configuration.GetSection("Logging"));
loggerFactory.AddDebug();

var logger = loggerFactory.CreateLogger<Startup>();
logger.LogInformation($"Runing in environment {env.EnvironmentName}");
```
4. Remove the loggerFactory configuration statements from the Configure method (2 firsts)

**Comment by bricelam** [comment]
This looks familiar. What version(s) of the EF packages are you using?

**Comment by mdarefull** [comment]
I have the latest stable updates of every nuget package, so they should be:
1.1.1 and 1.1.0 for Microsoft.EntityFrameworkCore.Tools

**Comment by MikeInSwitzerland** [comment]
It's now February 2018, and this issue is live and kicking with VS2017 v15.5.6 and EntityFramework Core v2.0.1.

**Comment by smitpatel** [comment]
@MikeInSwitzerland - This issue was fixed in EF Core 2.0 release. You may be seeing similar error message but it could be due to different bug. Please file a new issue with repro steps so that we can investigate.

**Comment by MikeInSwitzerland** [comment]
@smitpatel  Thank you so much for taking the time to reply to my comment... but as I said, I had actually installed Core 2.0 today.  What fixed my problem was to use VS2017's function to clear the GitHub cache.  But these error messages are so useless, and give you no clue about what to do, or how to solve them...  you don't know if it's going to take you an hour or a month to get around such problems...  As a developer, used to the cosy, reliable world of regular ASP.Net, it scares me....
